src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -225,11 +260,16 @@ class CoreWorkerDirectTaskSubmitter {
 
   struct SchedulingKeyEntry {
     // Keep track of pending worker lease requests to the raylet.
-    std::pair<std::shared_ptr<WorkerLeaseInterface>, TaskID> pending_lease_request =
-        std::make_pair(nullptr, TaskID::Nil());
+    std::deque<std::pair<std::shared_ptr<WorkerLeaseInterface>, TaskID>>
+        pending_lease_requests =
+            std::deque<std::pair<std::shared_ptr<WorkerLeaseInterface>, TaskID>>();
     // Tasks that are queued for execution. We keep an individual queue per
     // scheduling class to ensure fairness.
     std::deque<TaskSpecification> task_queue = std::deque<TaskSpecification>();
+    // Keep a queue of tasks that have not been used yet to request a new worker from the
+    // raylet.
+    std::deque<TaskSpecification> tasks_to_request_workers =
```

Comment:
Yes, that's right! With the current implementation of the RequestNewWorker RPC, if we request a new worker with a task ID that was used before, the Raylet will report an error and bad things will happen.

Overriding the TaskID sounds like a good idea, however my guess is that I would need to use a Non-Repeating Pseudo-Random generator to do that? Otherwise, even randomly generated TaskIDs could be duplicates.